chore: Typescript Cleanup Part 3 of 3 for New error errors after Version change

v-prachin opened the pull request:
#### Details

Typescript Cleanup Part 3 of 3 
We have total 90 errors remaining in 53 files.

##### Motivation
Resolved errors in all these files:

1. src/tests/unit/tests/DetailsView/components/adhoc-static-test-view.test.tsx (1)
2. src/tests/unit/tests/DetailsView/components/adhoc-tab-stops-test-view.test.tsx (1)
3. src/tests/unit/tests/DetailsView/components/iframe-warning.test.tsx (2)
4. src/tests/unit/tests/DetailsView/components/issues-table.test.tsx (1)
5. src/tests/unit/tests/DetailsView/details-view-container.test.tsx (2)
6. src/tests/unit/tests/injected/adapters/resolution-creator.test.ts (1)
7. src/tests/unit/tests/reports/package/axe-results-report.test.ts	(1)
8. src/tests/unit/tests/background/target-tab-controller.test.ts (1)
9. src/tests/unit/tests/background/stores/persistent-store.test.ts	(1)
10. src/tests/unit/tests/DetailsView/components/adhoc-issues-test-view.test.tsx(1)
11. src/tests/unit/tests/background/initial-visualization-store-data-generator.test.ts (2)
12. src/tests/unit/tests/reports/components/inline-image.test.tsx (1)

Number of errors covered by me as part of this user story: **15**


**Incorporated changes from Saanica's PR related to Feature 3**

1. src/tests/unit/tests/common/message-creators/assessment-card-selection-message-creator.test.ts
2. src/tests/unit/tests/common/message-creators/card-selection-message-creator.test.ts
3. src/tests/unit/tests/common/message-creators/needs-review-card-selection-message-creator.test.ts
4. src/tests/unit/tests/common/notification-creator.test.ts
5. src/tests/unit/tests/common/telemetry-data-factory.test.ts
6. src/tests/unit/tests/DetailsView/actions/details-view-action-message-creator.test.ts
7. src/tests/unit/tests/DetailsView/actions/tab-stop-requirement-action-message-creator.test.ts

Number of errors covered by Saanica: **14**


Thus, total number of errors covered as part of this user story: **29**


##### Context

#### Pull request checklist
<!-- If a checklist item is not applicable to this change, write "n/a" in the checkbox -->
- [x] Addresses an existing issue: #(https://github.com/microsoft/accessibility-insights-web/pull/6611)
- [x] Ran `yarn fastpass`
- [x] Added/updated relevant unit test(s) (and ran `yarn test`)
- [x] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] PR title *AND* final merge commit title both start with a semantic tag (`fix:`, `chore:`, `feat(feature-name):`, `refactor:`). See `CONTRIBUTING.md`.
- [n/a] (UI changes only) Added screenshots/GIFs to description above
- [n/a] (UI changes only) Verified usability with NVDA/JAWS

v-prachin committed:
Merge branch 'microsoft:main' into typescript-feature3-part3

v-prachin committed:
chore: Typescript Cleanup Part 2 of 3 for rule: ruleStub & launchPanelType: panelType (#7289)

#### Details

Typescript Cleanup Part 2 of 3 
We have total 140 errors in 104 files. Thus, targeting each error and
fixing it.

##### Motivation

Errors observed in below files observed after removing deprecated
properties from tsconfig file:
1.**rule: ruleStub**  
src/tests/unit/tests/injected/element-based-view-model-creator.test.ts
(9)
scanNodeResults consists of unified results as well as ruleStub. So,
using scanNodeResults to resolve the issue.

2.**launchPanelType: panelType**  
src/popup/actions/popup-action-message-creator.ts(1)

src/tests/unit/tests/popup/actions/popup-action-message-creator.test.ts(1)
Added new type LaunchPanelTelemetryData in
extension-telemetry-events.ts, to resolve the error for launchPanelType:
panelType

Thus, total number of errors covered as part of this user story: **11**


##### Context

#### Pull request checklist
<!-- If a checklist item is not applicable to this change, write "n/a"
in the checkbox -->
- [x] Addresses an existing issue:
#(https://github.com/microsoft/accessibility-insights-web/pull/6611)
- [x] Ran `yarn fastpass`
- [x] Added/updated relevant unit test(s) (and ran `yarn test`)
- [x] Verified code coverage for the changes made. Check coverage report
at: `<rootDir>/test-results/unit/coverage`
- [x] PR title *AND* final merge commit title both start with a semantic
tag (`fix:`, `chore:`, `feat(feature-name):`, `refactor:`). See
`CONTRIBUTING.md`.
- [n/a] (UI changes only) Added screenshots/GIFs to description above
- [n/a] (UI changes only) Verified usability with NVDA/JAWS

v-prachin committed:
Typescript Cleanup of errors after version change

v-prachin committed:
chore: Typescript Cleanup Part 2 of 3 for order, x:left, pageTitle, requirement, titleSize="title & defaultViewport: null (#7280)

#### Details

Typescript Cleanup Part 2 of 3 
We have total 140 errors in 104 files. Thus, targeting each error and
fixing it.

##### Motivation

Errors observed in below files observed after removing deprecated
properties from tsconfig file:
1.**order:0**  

src/issue-filing/services/null-issue-filing-service/null-issue-filing-service.tsx
Removed this property as NullIssueFilingService has this order property
which seems to be not consumed, as per my understanding. As checked for
the references order: and couldn't observe anything relevant to
NullIssueFilingService.

2.**titleSize="title"**  

src/reports/components/report-sections/combined-report-rules-only-sections.tsx
(1)
Removed this property as CombinedReportResultSectionTitle in other files
only has three entities in it, outcomeCount,
outcomeType, title. That file is combined-report-failed-section.tsx

3.**defaultViewport: null**  
src/tests/end-to-end/common/browser-factory.ts (1) 
Removed this property as defaultViewport has only one reference in the
entire solution and that is in brower-factory.ts

4.**requirement: requirementKey**  
src/tests/unit/tests/background/stores/assessment-store.test.ts (3) 
Added appropriate interface for requirement & selector properties.

5.**pageTitle**  

src/tests/unit/tests/reports/components/fast-pass-report-automated-checks-results.test.tsx
(1)
src/tests/unit/tests/reports/components/fast-pass-report.test.tsx (1) 
Removed this property as it was not needed in the test file.

6.**x:left**
src/injected/visualization/drawer-utils.ts (1)
Removed this property as BoundingRect already has left and top
properties defined.

Thus, total number of errors covered as part of this user story: **9**


##### Context

#### Pull request checklist
<!-- If a checklist item is not applicable to this change, write "n/a"
in the checkbox -->
- [x] Addresses an existing issue:
#(https://github.com/microsoft/accessibility-insights-web/pull/6611)
- [x] Ran `yarn fastpass`
- [x] Added/updated relevant unit test(s) (and ran `yarn test`)
- [x] Verified code coverage for the changes made. Check coverage report
at: `<rootDir>/test-results/unit/coverage`
- [x] PR title *AND* final merge commit title both start with a semantic
tag (`fix:`, `chore:`, `feat(feature-name):`, `refactor:`). See
`CONTRIBUTING.md`.
- [n/a] (UI changes only) Added screenshots/GIFs to description above
- [n/a] (UI changes only) Verified usability with NVDA/JAWS

JeevaniChinthala committed:
chore: Typescript Cleanup- Part 2 for isSelected,Requirement,fixInstructionProcessor (#7256)

#### Details

As part of Typescript migration added/removed the error causing fields.

##### Motivation

TypeScript 5.0 has marked some options as deprecated. We can temporarily
override these deprecations, but the documented plan is for the ability
to override the flag to go away in TypeScript 5.5

##### Context

1. **Requirement**

- **Error:** Type '{ test: VisualizationType; requirement: string;
telemetry: AssessmentTelemetryData; }' is not assignable to type
'ToggleActionPayload'.
- Object literal may only specify known properties, and 'requirement'
does not exist in type 'ToggleActionPayload'.
- In AssessmentToggleActionPayload interface the methods
enableFastPassVisualHelperWithoutScan and
rescanVisualizationWithoutTelemetry uses the requirement property which
was an optional property. To avoid null check error a new interface
named RequirementToggleActionPayload was introduced, in which the
requirement property is declared as either string or undefined.

2. **isSelected**

- **Error:** Type '{ isSelected: boolean; status: InstanceResultStatus;
ruleId: string; uid: string; identifiers: { conciseName: string;
identifier: string; 'css-selector': string; target: Target; };
descriptors: { ...; }; resolution: { ...; }; rule: { ...; }; }' is not
assignable to type 'ScanNodeResult'.
- Object literal may only specify known properties, and 'isSelected'
does not exist in type 'ScanNodeResult'.
- Added isSelected in ScanNodeResult in order to get required data.

3.   **fixInstructionProcessor**

- **Error:** Type'{ deps: InstanceDetailsGroupDeps;
fixInstructionProcessor: FixInstructionProcessor; recommendColor:
RecommendColor; rule: CardRuleResult; userConfigurationStoreData: null;
targetAppInfo: { ...; }; cardSelectionMessageCreator:
CardSelectionMessageCreator; }' is not assignable to type
'InstanceDetailsGroupProps'.
- Object literal may only specify known properties, and
'fixInstructionProcessor' does not exist in type
'InstanceDetailsGroupProps'.
- Upon verifying in UI, fixInstructionProcessor is defined in 'deps'
property. So, removed from code.

#### Pull request checklist
<!-- If a checklist item is not applicable to this change, write "n/a"
in the checkbox -->
- [x] Addresses an existing issue:
#(https://github.com/microsoft/accessibility-insights-web/pull/6611)
- [ ] Ran `yarn fastpass`
- [x] Added/updated relevant unit test(s) (and ran `yarn test`)
- [x] Verified code coverage for the changes made. Check coverage report
at: `<rootDir>/test-results/unit/coverage`
- [ ] PR title *AND* final merge commit title both start with a semantic
tag (`fix:`, `chore:`, `feat(feature-name):`, `refactor:`). See
`CONTRIBUTING.md`.
- [ ] (UI changes only) Added screenshots/GIFs to description above
- [ ] (UI changes only) Verified usability with NVDA/JAWS

v-viyada commented on src/tests/unit/tests/injected/adapters/resolution-creator.test.ts:
```diff
@@ -62,8 +62,6 @@ describe('ResolutionCreator', () => {
                 ],
                 all: [{ id: null, message: 'all 1 message', data: null }],
                 none: [{ id: null, message: 'none 1 message', data: null }],
-                html: 'test html',
```
why this is removed?

v-prachin replied:
NodeResolutionData Inteface does not consist of html property, checked git history. Thus, removed the property.

v-viyada commented on yarn.lock:
```diff
@@ -3339,7 +3339,7 @@ __metadata:
     eslint-plugin-import: ^2.29.1
     eslint-plugin-react: ^7.33.2
     eslint-plugin-security: ^1.7.1
-    express: ^4.18.2
```
why there are yarn lock changes.

v-prachin replied:
In accessibility insights Main repo this version is upgraded to 4.19.2. The branch typescript-part5 was not upto date with main thus this change is showing.

v-prachin replied:
For package.json and yarn lock taken latest changes from typescript-part5 branch and pasted as it is in these files. So these changes are not due to fixing errors

JeevaniChinthala committed:
chore: TypeScript Cleanup 5.0 Part 4 for Errors - includeStepsText,featureFlagStoreData,deps,requirementState,outline, (#7292)

#### Details

As part of Typescript migration added/removed the error causing fields.

##### Motivation

TypeScript 5.0 has marked some options as deprecated. We can temporarily
override these deprecations, but the documented plan is for the ability
to override the flag to go away in TypeScript 5.5

##### Context

1. **includeStepsText={false}**

- **Error:** No overload matches this call.Type '{
needsReviewCardsViewData: CardsViewModel;
needsReviewCardSelectionMessageCreator: CardSelectionMessageCreator;
automatedChecksCardsViewData: CardsViewModel }' is not assignable to
type 'IntrinsicAttributes &
IntrinsicClassAttributes<AssessmentIssuesTestView> &
Readonly<AssessmentIssuesTestViewProps> & Readonly<...>'
- Property 'includeStepsText' does not exist on type
'IntrinsicAttributes &
IntrinsicClassAttributes<AssessmentIssuesTestView> &
Readonly<AssessmentIssuesTestViewProps>
- In file
src/DetailsView/components/default-test-view-container-provider.tsx
getting error for includeStepsText. Added the property in reference
file.

2. **requirementState={requirementState}**

- **Error:** Type '{ deps: AdhocTabStopsTestViewDeps;
failureInstanceState: FailureInstanceState; requirementState:
TabStopRequirementState; }' is not assignable to type
'IntrinsicAttributes & TabStopsFailedInstancePanelProps & { children?:
ReactNode; }'.
- Property 'requirementState' does not exist on type
'IntrinsicAttributes & TabStopsFailedInstancePanelProps & { children?:
ReactNode; }'.
- As per the previous commits, someone removed from the reference
interface/type but didn't remove from actual usage.
3.  **featureFlagStoreData={featureFlagStoreData}**

- **Error:** Type '{ deps: DetailsViewLeftNavDeps; featureFlagStoreData:
FeatureFlagStoreData; pivotKey: DetailsViewPivotType; }' is not
assignable to type 'IntrinsicAttributes &
IntrinsicClassAttributes<Switcher> & Readonly<SwitcherProps> &
Readonly<...>'.
- Property 'featureFlagStoreData' does not exist on type
'IntrinsicAttributes & IntrinsicClassAttributes<Switcher>
- As per the previous commits, someone removed from the reference
interface/type but didn't remove from actual usage.

4. **Deps**

- **Error:** Type '{ deps: ReportExportComponentDeps; isOpen: boolean;
htmlFileName: string; jsonFileName: string; description: string;
htmlExportData: string; jsonExportData: string; ... 9 more ...;
exportResultsClickedTelemetry: (reportExportFormat: ReportExportFormat,
selectedServiceKey: ReportExportServiceKey, event: MouseEvent<.....' is
not assignable to type 'IntrinsicAttributes & ExportDialogProps & {
children?: ReactNode; }'.
- Property 'deps' does not exist on type 'IntrinsicAttributes &
ExportDialogProps & { children?: ReactNode; }'.
- As per the previous commits, someone removed from the reference
interface/type but didn't remove from actual usage.

5. **outline: '3px dashed'**

- **Error:** Type '{ fontColor: string; background: string; text:
string; fontWeight: string; fontSize: string; outline: string; }' is not
assignable to type 'TextBoxConfig'.
- Object literal featureFlagData={featureFlags}may only specify known
properties, and 'outline' does not exist in type 'TextBoxConfig'.
- As per the previous commits, someone removed from the reference
interface/type but didn't remove from actual usage.

6. **featureFlagData={featureFlags}**

- **Error:** Type '{ deps: LaunchPanelHeaderDeps; popupWindow: Window;
header: this; featureFlagData: FeatureFlagStoreData; }' is not
assignable to type 'IntrinsicAttributes & HamburgerMenuButtonProps & {
children?: ReactNode; }'.
- Property 'featureFlagData' does not exist on type 'IntrinsicAttributes
& HamburgerMenuButtonProps & { children?: ReactNode; }'.
- As per the previous commits, someone removed from the reference
interface/type but didn't remove from actual usage.

7. **isContextMenuVisible**

- **Error:** Argument of type '{ target: any; isContextMenuVisible:
boolean; }' is not assignable to parameter of type
'LaunchPanelHeaderState | ((prevState: Readonly<LaunchPanelHeaderState>,
props: Readonly<LaunchPanelHeaderProps>) => LaunchPanelHeaderState |
Pick<...>) | Pick<...>'.
- Object literal may only specify known properties, and
'isContextMenuVisible' does not exist in type 'LaunchPanelHeaderState
- As per the previous commits, someone removed from the reference
interface/type but didn't remove from actual usage.

8. **featureFlagStoreData**

- **Error:** Property 'featureFlagStoreData' does not exist on type
'IntrinsicAttributes & IntrinsicClassAttributes<FastPassReportSummary>
- As per the previous commits, someone removed from the reference
interface/type but didn't remove from actual usage.

9. **deps:null**

- **Error:** Argument of type '{ id: string; header: JSX.Element;
content: JSX.Element; headingLevel: number; deps: null; }' is not
assignable to parameter of type 'ReportCollapsibleContainerProps'.
- Object literal may only specify known properties, and 'deps' does not
exist in type 'ReportCollapsibleContainerProps'.
- As per the previous commits, someone removed from the reference
interface/type but didn't remove from actual usage.

10.  **testName: 'test-name'**

- **Error:** Type '{ key: string; testName: string; rules: string[];
testType: VisualizationType; }' is not assignable to type
'Readonly<Partial<RuleAnalyzerConfiguration>>'.
- Object literal may only specify known properties, and 'testName' does
not exist in type 'Readonly<Partial<RuleAnalyzerConfiguration>>'.

11. **openGettingStartedDialog**

- **Error:** Type '{ deps: LaunchPanelHeaderDeps; title: string;
subtitle: string; openGettingStartedDialog: any; openFeedbackDialog:
any; popupWindow: Window; featureFlags: FeatureFlagStoreData;
openAdhocToolsPanel: () => void; dropdownClickHandler:
DropdownClickHandler; }' is not assignable to type
'LaunchPanelHeaderProps'.
- Object literal may only specify known properties, and
'openGettingStartedDialog' does not exist in type
'LaunchPanelHeaderProps'.
- As per the previous commits, someone removed from the reference
interface/type but didn't remove from actual usage.

12. **deps**

- **Error:** Type '{ deps: { collapsibleControl:
(collapsibleControlProps: ReportCollapsibleContainerProps) =>
JSX.Element; }; failed: { url: string; numFailures: number;
reportLocation: string; }[]; passed: { ...; }[]; unscannable: { ...;
}[]; }' is not assignable to type 'SummaryScanResults'.
- Object literal may only specify known properties, and 'deps' does not
exist in type 'SummaryScanResults'.
- As per the previous commits, someone removed from the reference
interface/type but didn't remove from actual usage.


#### Pull request checklist
<!-- If a checklist item is not applicable to this change, write "n/a"
in the checkbox -->
- [x] Addresses an existing issue:
#[0000](https://github.com/microsoft/accessibility-insights-web/pull/6611)
- [ ] Ran `yarn fastpass`
- [ ] Added/updated relevant unit test(s) (and ran `yarn test`)
- [x] Verified code coverage for the changes made. Check coverage report
at: `<rootDir>/test-results/unit/coverage`
- [x ] PR title *AND* final merge commit title both start with a
semantic tag (`fix:`, `chore:`, `feat(feature-name):`, `refactor:`). See
`CONTRIBUTING.md`.
- [ ] (UI changes only) Added screenshots/GIFs to description above
- [ ] (UI changes only) Verified usability with NVDA/JAWS

---------

Co-authored-by: Prachi Naigaonkar <v-prachin@microsoft.com>

dependabot[bot] committed:
chore(deps-dev): Bump esbuild from 0.19.11 to 0.20.2 (#7278)

Bumps [esbuild](https://github.com/evanw/esbuild) from 0.19.11 to
0.20.2.
<details>
<summary>Release notes</summary>
<p><em>Sourced from <a
href="https://github.com/evanw/esbuild/releases">esbuild's
releases</a>.</em></p>
<blockquote>
<h2>v0.20.2</h2>
<ul>
<li>
<p>Support TypeScript experimental decorators on <code>abstract</code>
class fields (<a
href="https://redirect.github.com/evanw/esbuild/issues/3684">#3684</a>)</p>
<p>With this release, you can now use TypeScript experimental decorators
on <code>abstract</code> class fields. This was silently compiled
incorrectly in esbuild 0.19.7 and below, and was an error from esbuild
0.19.8 to esbuild 0.20.1. Code such as the following should now work
correctly:</p>
<pre lang="ts"><code>// Original code
const log = (x: any, y: string) =&gt; console.log(y)
abstract class Foo { @log abstract foo: string }
new class extends Foo { foo = '' }
<p>// Old output (with --loader=ts
--tsconfig-raw={&amp;quot;compilerOptions&amp;quot;:{&amp;quot;experimentalDecorators&amp;quot;:true}})
const log = (x, y) =&gt; console.log(y);
class Foo {
}
new class extends Foo {
foo = &quot;&quot;;
}();</p>
<p>// New output (with --loader=ts
--tsconfig-raw={&amp;quot;compilerOptions&amp;quot;:{&amp;quot;experimentalDecorators&amp;quot;:true}})
const log = (x, y) =&gt; console.log(y);
class Foo {
}
__decorateClass([
log
], Foo.prototype, &quot;foo&quot;, 2);
new class extends Foo {
foo = &quot;&quot;;
}();
</code></pre></p>
</li>
<li>
<p>JSON loader now preserves <code>__proto__</code> properties (<a
href="https://redirect.github.com/evanw/esbuild/issues/3700">#3700</a>)</p>
<p>Copying JSON source code into a JavaScript file will change its
meaning if a JSON object contains the <code>__proto__</code> key. A
literal <code>__proto__</code> property in a JavaScript object literal
sets the prototype of the object instead of adding a property named
<code>__proto__</code>, while a literal <code>__proto__</code> property
in a JSON object literal just adds a property named
<code>__proto__</code>. With this release, esbuild will now work around
this problem by converting JSON to JavaScript with a computed property
key in this case:</p>
<pre lang="js"><code>// Original code
import data from
'data:application/json,{&quot;__proto__&quot;:{&quot;fail&quot;:true}}'
if (Object.getPrototypeOf(data)?.fail) throw 'fail'
<p>// Old output (with --bundle)
(() =&gt; {
//
&lt;data:application/json,{&quot;<strong>proto</strong>&quot;:{&quot;fail&quot;:true}}&gt;
var json_proto_fail_true_default = { <strong>proto</strong>: { fail:
true } };</p>
<p>// entry.js
if (Object.getPrototypeOf(json_proto_fail_true_default)?.fail)
throw &quot;fail&quot;;
})();</p>
<p></code></pre></p>
</li>
</ul>
<!-- raw HTML omitted -->
</blockquote>
<p>... (truncated)</p>
</details>
<details>
<summary>Changelog</summary>
<p><em>Sourced from <a
href="https://github.com/evanw/esbuild/blob/main/CHANGELOG.md">esbuild's
changelog</a>.</em></p>
<blockquote>
<h2>0.20.2</h2>
<ul>
<li>
<p>Support TypeScript experimental decorators on <code>abstract</code>
class fields (<a
href="https://redirect.github.com/evanw/esbuild/issues/3684">#3684</a>)</p>
<p>With this release, you can now use TypeScript experimental decorators
on <code>abstract</code> class fields. This was silently compiled
incorrectly in esbuild 0.19.7 and below, and was an error from esbuild
0.19.8 to esbuild 0.20.1. Code such as the following should now work
correctly:</p>
<pre lang="ts"><code>// Original code
const log = (x: any, y: string) =&gt; console.log(y)
abstract class Foo { @log abstract foo: string }
new class extends Foo { foo = '' }
<p>// Old output (with --loader=ts
--tsconfig-raw={&amp;quot;compilerOptions&amp;quot;:{&amp;quot;experimentalDecorators&amp;quot;:true}})
const log = (x, y) =&gt; console.log(y);
class Foo {
}
new class extends Foo {
foo = &quot;&quot;;
}();</p>
<p>// New output (with --loader=ts
--tsconfig-raw={&amp;quot;compilerOptions&amp;quot;:{&amp;quot;experimentalDecorators&amp;quot;:true}})
const log = (x, y) =&gt; console.log(y);
class Foo {
}
__decorateClass([
log
], Foo.prototype, &quot;foo&quot;, 2);
new class extends Foo {
foo = &quot;&quot;;
}();
</code></pre></p>
</li>
<li>
<p>JSON loader now preserves <code>__proto__</code> properties (<a
href="https://redirect.github.com/evanw/esbuild/issues/3700">#3700</a>)</p>
<p>Copying JSON source code into a JavaScript file will change its
meaning if a JSON object contains the <code>__proto__</code> key. A
literal <code>__proto__</code> property in a JavaScript object literal
sets the prototype of the object instead of adding a property named
<code>__proto__</code>, while a literal <code>__proto__</code> property
in a JSON object literal just adds a property named
<code>__proto__</code>. With this release, esbuild will now work around
this problem by converting JSON to JavaScript with a computed property
key in this case:</p>
<pre lang="js"><code>// Original code
import data from
'data:application/json,{&quot;__proto__&quot;:{&quot;fail&quot;:true}}'
if (Object.getPrototypeOf(data)?.fail) throw 'fail'
<p>// Old output (with --bundle)
(() =&gt; {
//
&lt;data:application/json,{&quot;<strong>proto</strong>&quot;:{&quot;fail&quot;:true}}&gt;
var json_proto_fail_true_default = { <strong>proto</strong>: { fail:
true } };</p>
<p>// entry.js
if (Object.getPrototypeOf(json_proto_fail_true_default)?.fail)
throw &quot;fail&quot;;
})();
</code></pre></p>
</li>
</ul>
<!-- raw HTML omitted -->
</blockquote>
<p>... (truncated)</p>
</details>
<details>
<summary>Commits</summary>
<ul>
<li><a
href="https://github.com/evanw/esbuild/commit/617eddaa32b7649ad23ddd15257816df3f0f544c"><code>617edda</code></a>
publish 0.20.2 to npm</li>
<li><a
href="https://github.com/evanw/esbuild/commit/4780075ace9f5f59b61018326ec57da624a7d7a2"><code>4780075</code></a>
fix <a
href="https://redirect.github.com/evanw/esbuild/issues/3700">#3700</a>:
json loader preserves <code>__proto__</code> keys</li>
<li><a
href="https://github.com/evanw/esbuild/commit/30bed2d2d7572b58399a6c07643878f3377800fe"><code>30bed2d</code></a>
better errors for invalid js decorator syntax</li>
<li><a
href="https://github.com/evanw/esbuild/commit/300eeb711261144a70ab33221fe6a1313b309348"><code>300eeb7</code></a>
ts: allow non-null assertions in js decorators</li>
<li><a
href="https://github.com/evanw/esbuild/commit/4d997d948579b2e4e9bd3bf820a0108b58b3732b"><code>4d997d9</code></a>
fix <a
href="https://redirect.github.com/evanw/esbuild/issues/3698">#3698</a>:
yarn pnp edge case with <code>tsconfig.json</code></li>
<li><a
href="https://github.com/evanw/esbuild/commit/cf42954760e666c048a866589f43ef937f9ec1e6"><code>cf42954</code></a>
resolver: improve some debug logging</li>
<li><a
href="https://github.com/evanw/esbuild/commit/b0765ae1323363fd61727df54e9e57c2b81bdd4b"><code>b0765ae</code></a>
fix some lints</li>
<li><a
href="https://github.com/evanw/esbuild/commit/dfa62069192b1c49918b2dfe4b8c10278361ae34"><code>dfa6206</code></a>
fix some comments (closes <a
href="https://redirect.github.com/evanw/esbuild/issues/3683">#3683</a>)</li>
<li><a
href="https://github.com/evanw/esbuild/commit/ae5cc175165551016f8e19475c4ac9872e60f247"><code>ae5cc17</code></a>
fix <a
href="https://redirect.github.com/evanw/esbuild/issues/3684">#3684</a>:
<code>abstract</code> experimental decorators</li>
<li><a
href="https://github.com/evanw/esbuild/commit/c809af050a74f022d9cf61c66e13365434542420"><code>c809af0</code></a>
fix <a
href="https://redirect.github.com/evanw/esbuild/issues/2388">#2388</a>:
allow consuming types without dom types (<a
href="https://redirect.github.com/evanw/esbuild/issues/3679">#3679</a>)</li>
<li>Additional commits viewable in <a
href="https://github.com/evanw/esbuild/compare/v0.19.11...v0.20.2">compare
view</a></li>
</ul>
</details>
<br />


[![Dependabot compatibility
score](https://dependabot-badges.githubapp.com/badges/compatibility_score?dependency-name=esbuild&package-manager=npm_and_yarn&previous-version=0.19.11&new-version=0.20.2)](https://docs.github.com/en/github/managing-security-vulnerabilities/about-dependabot-security-updates#about-compatibility-scores)

Dependabot will resolve any conflicts with this PR as long as you don't
alter it yourself. You can also trigger a rebase manually by commenting
`@dependabot rebase`.

[//]: # (dependabot-automerge-start)
[//]: # (dependabot-automerge-end)

---

<details>
<summary>Dependabot commands and options</summary>
<br />

You can trigger Dependabot actions by commenting on this PR:
- `@dependabot rebase` will rebase this PR
- `@dependabot recreate` will recreate this PR, overwriting any edits
that have been made to it
- `@dependabot merge` will merge this PR after your CI passes on it
- `@dependabot squash and merge` will squash and merge this PR after
your CI passes on it
- `@dependabot cancel merge` will cancel a previously requested merge
and block automerging
- `@dependabot reopen` will reopen this PR if it is closed
- `@dependabot close` will close this PR and stop Dependabot recreating
it. You can achieve the same result by closing it manually
- `@dependabot show <dependency name> ignore conditions` will show all
of the ignore conditions of the specified dependency
- `@dependabot ignore this major version` will close this PR and stop
Dependabot creating any more for this major version (unless you reopen
the PR or upgrade to it yourself)
- `@dependabot ignore this minor version` will close this PR and stop
Dependabot creating any more for this minor version (unless you reopen
the PR or upgrade to it yourself)
- `@dependabot ignore this dependency` will close this PR and stop
Dependabot creating any more for this dependency (unless you reopen the
PR or upgrade to it yourself)


</details>

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

JeevaniChinthala committed:
chore: Typescript Cleanup- Part 3 for Test Files (#7291)

#### Details

As part of Typescript migration added/removed the error causing fields.

##### Motivation

TypeScript 5.0 has marked some options as deprecated. We can temporarily
override these deprecations, but the documented plan is for the ability
to override the flag to go away in TypeScript 5.5

##### Context

1.  **featureFlagStoreData**

- **Error:** Type '{ backLinkHandler: () => void;
diagnosticViewToggleFactory: DiagnosticViewToggleFactory;
featureFlagStoreData: FeatureFlagStoreData; }' is not assignable to type
'IntrinsicAttributes & AdHocToolsPanelProps & { children?: ReactNode;
}'.
Property 'featureFlagStoreData' does not exist on type
'IntrinsicAttributes & AdHocToolsPanelProps & { children?: ReactNode;
}'.

2. **avatarUrl: 'avatarUrl'**

- **Error:** Type '{ featureFlagStoreData: { 'test-flag': true; };
avatarUrl: string; tabClosed: boolean; deps: InteractiveHeaderDeps;
selectedPivot: DetailsViewPivotType.assessment; navMenu:
ReactFCWithDisplayName<...>; narrowModeStatus: NarrowModeStatus;
isSideNavOpen: false; setSideNavOpen: null; }' is not assignable to type
'InteractiveHeaderProps'.
- Object literal may only specify known properties, and 'avatarUrl' does
not exist in type 'InteractiveHeaderProps'.

3. **status: ManualTestStatus.FAIL**

- **Error:** Type '{ step: string; test:
VisualizationType.HeadingsAssessment; status: ManualTestStatus;
assessmentInstanceTableHandler: AssessmentInstanceTableHandler;
manualTestStepResultMap: { ...; }; assessmentsProvider:
AssessmentsProvider; featureFlagStoreData: FeatureFlagStoreData;
pathSnippetStoreData: { ...; }; }' is not assignable to type
'ManualTestStepViewProps'.
- Object literal may only specify known properties, and 'status' does
not exist in type 'ManualTestStepViewProps'.

4. **supportLinkHandler: null**

- **Error:** Type '{ deps: LaunchPanelHeaderDeps; title: string;
subtitle: string; supportLinkHandler: null; popupWindow: null;
featureFlags: null; openAdhocToolsPanel: null; dropdownClickHandler:
null; }' is not assignable to type 'LaunchPanelHeaderProps'.
- Object literal may only specify known properties, and
'supportLinkHandler' does not exist in type 'LaunchPanelHeaderProps'.
- In the reference file the supportLinkHandler is removed and
dropdownClickHandler is already defined in LaunchPanelHeaderDeps

5. **browserAdapter: browserAdapterMock.object**

- **Error:** Type '{ diagnosticViewClickHandler:
DiagnosticViewClickHandler; popupViewControllerHandler:
PopupViewControllerHandler; launchPanelHeaderClickHandler:
LaunchPanelHeaderClickHandler; browserAdapter: BrowserAdapter; }' is not
assignable to type 'PopupHandlers'.
- Object literal may only specify known properties, and 'browserAdapter'
does not exist in type 'PopupHandlers'.

6. **targetAppInfo**

- **Error:** Type '{ targetAppInfo: { name: string; url: string; };
description: string; environmentInfo: { browserSpec: string;
extensionVersion: string; axeCoreVersion: string; }; toUtcString: (date:
Date) => string; scanMetadata: ScanMetadata; }' is not assignable to
type 'DetailsSectionProps'.
- Object literal may only specify known properties, and 'targetAppInfo'
does not exist in type 'DetailsSectionProps'.
- Removed 'environmentInfo' property as there are no references defined
in 'DetailsSectionProps' type

7. **outcomeType: outcomeType,**

- **Error:** Type '{ customCongratsContinueInvestigatingMessage: string;
outcomeType: InstanceOutcomeType; }' is not assignable to type
'NoFailedInstancesCongratsDeps'.
- Object literal may only specify known properties, and 'outcomeType'
does not exist in type 'NoFailedInstancesCongratsDeps'.

8. **pageTitle**

- **Error:** Type '{ deps: ResultSectionContentDeps;
fixInstructionProcessor: FixInstructionProcessor; recommendColor:
RecommendColor; ... 13 more ...; sectionHeadingLevel: 3; }' is not
assignable to type 'SectionProps'.
- Object literal may only specify known properties, and 'pageTitle' does
not exist in type 'SectionProps'
- pageUrl,toolData,scanResult doesn't exist in type 'SectionProps'.
Hence removed.

9.  **deps:null**

- **Error:** Type '{ [x: string]: string | number | boolean |
JSX.Element; id: string; header: JSX.Element; content: JSX.Element;
headingLevel: number; deps: null; isExpanded: boolean; }' is not
assignable to type 'ReportCollapsibleContainerProps'.
- Object literal may only specify known properties, and 'deps' does not
exist in type 'ReportCollapsibleContainerProps'.
- isExpanded doesn't exist in type 'ReportCollapsibleContainerProps'.
Hence removed.


10.  **{ name: 'Any', key: 'any' }**

- **Error :** Type '{ name: string; key: string; }' is not assignable to
type 'IRequirementSubsetForSummary'.
- Object literal may only specify known properties, and 'key' does not
exist in type 'IRequirementSubsetForSummary'.

11. **cardSelectionMessageCreator**

- **Error:** Type '{ serviceName: string; axeVersion: string; userAgent:
string; browserResolution: string; scanDetails: ScanSummaryDetails;
results: CombinedReportResults; cardSelectionMessageCreator:
CardSelectionMessageCreator; }' is not assignable to type
'CombinedReportParameters'.
- Object literal may only specify known properties, and
'cardSelectionMessageCreator' does not exist in type
'CombinedReportParameters'.


#### Pull request checklist
<!-- If a checklist item is not applicable to this change, write "n/a"
in the checkbox -->
- [x] Addresses an existing issue:
#[[0000](https://github.com/microsoft/accessibility-insights-web/pull/6611)](https://github.com/microsoft/accessibility-insights-web/pull/6611)
- [ ] Ran `yarn fastpass`
- [x] Added/updated relevant unit test(s) (and ran `yarn test`)
- [x] Verified code coverage for the changes made. Check coverage report
at: `<rootDir>/test-results/unit/coverage`
- [ ] PR title *AND* final merge commit title both start with a semantic
tag (`fix:`, `chore:`, `feat(feature-name):`, `refactor:`). See
`CONTRIBUTING.md`.
- [ ] (UI changes only) Added screenshots/GIFs to description above
- [ ] (UI changes only) Verified usability with NVDA/JAWS

---------

Co-authored-by: Madalyn <3230904+madalynrose@users.noreply.github.com>

v-prachin committed:
PR review comments changes, New files, Saanica's PR changes

v-prachin committed:
Merge pull request #4 from v-prachin/main

chore: Merging Main into Feature branch
